Editor/UI/ProjectAuditorWindow.cs

```diff
@@ -419,22 +437,252 @@ private void DrawCallHierarchy(CallTreeNode callTree)
+            int selectedCount = 0;
+            foreach (var name in dict.Keys)
+            {
+                if (selectionDict[name] != dict[name])
+                    continue;
+
+                selectedCount++;
+            }
+            
+            // If we've just added all the item names we have everything selected
+            // Note we don't compare against the names array directly as this contains the 'all' versions
+            if (selectedCount == dict.Keys.Count)
+                return "All";
+
+            // Add all the individual items were we haven't already added the group
+            List<string> individualItems = new List<string>();
+            foreach (var name in selectionDict.Keys)
+            {
+                int selectionCount = selectionDict[name];
+                if (selectionCount <= 0)
+                    continue;
+                int itemCount = dict[name];
+                if (itemCount == 1)
+                    individualItems.Add(name);
+                else if (selectionCount != itemCount)
+                    individualItems.Add(string.Format("{0} ({1} of {2})", name, selectionCount, itemCount));
+                else
+                    individualItems.Add(string.Format("{0} (All)", name));
+            }
+
+            // Maintain alphabetical order
+            individualItems.Sort(CompareUINames);
+
+            if (individualItems.Count == 0)
+                return "None";
+
+            string selectedText = string.Join(", ", individualItems.ToArray());
+            return selectedText;
+        }
+        
+        private int CompareUINames(string a, string b)
+        {
+            string[] aTokens = a.Split(':');
+            string[] bTokens = b.Split(':');
+
+            if (aTokens.Length > 1 && bTokens.Length > 1)
+            {
+                var aThreadName = aTokens[0].Trim();
+                var bThreadName = bTokens[0].Trim();
+
+                if (aThreadName == bThreadName)
+                {
+                    string aThreadIndex = aTokens[1].Trim();
+                    string bThreadIndex = bTokens[1].Trim();
+
+                    if (aThreadIndex == "All" && bThreadIndex != "All")
+                        return -1;
+                    if (aThreadIndex != "All" && bThreadIndex == "All")
+                        return 1;
+
+                    int aGroupIndex;
+                    if (int.TryParse(aThreadIndex, out aGroupIndex))
+                    {
+                        int bGroupIndex;
+                        if (int.TryParse(bThreadIndex, out bGroupIndex))
+                        {
+                            return aGroupIndex.CompareTo(bGroupIndex);
+                        }
+                    }
+                }
+            }
+
+            return a.CompareTo(b);
+        }
+        
+        private void DrawSelectedText(string text)
+        {
+#if UNITY_2019_1_OR_NEWER
+            GUIStyle treeViewSelectionStyle = "TV Selection";
```

Comment:
These appear to be undocumented GUIStyles that are in the built-in Unity themes (DarkSkin and LightSkin) which presumably only went into Unity in 2019.1. The only reference I can find for them (other than the code in Profile Analyzer this is based on) is here: 
https://github.com/Unity-Technologies/UnityCsReference/blob/master/Editor/Mono/GUI/TreeView/TreeViewGUI.cs

The effect of these GUIStyles is to make the summary text of the selected assemblies/items to be drawn with a blue background (in 2019.1 or newer), as if the text has been selected or highlighted. It just renders as normal text in older versions of Unity. This is consistent with Profile Analyzer, but I can only speculate as to why the text is drawn in this way - perhaps as a hint that the selection (and thus the summary string) can be edited by the user, by pressing the select button?

I assume this meets the UI team's approval, but I'm equally happy to remove it and render the summary as plain text if you'd prefer. My inclination is to leave it as-is because it was presumably implemented that way for a reason, but if you'd prefer I removed it, I will.

Reply:
Fair enough. Let's leave it.